### ISSUE TITLE
allow relative and absolute globs

### ISSUE THREAD

**Issue by plexigras** [eslint/eslint#11003]
**The version of ESLint you are using.**

`5.7.0`

**The problem you want to solve.**

i ran into this while doing

```
eslint --config=./configs/eslint.js test
```

but globs like `../test/*` and even absolute paths like `/**/test/*` and `process.cwd() + '/test/*'` are prohibited

https://github.com/eslint/eslint/blob/master/lib/config/config-ops.js#L364

**Your take on the correct solution to problem.**

all i see that would need to be changed is to also check against the absolute path.

https://github.com/eslint/eslint/blob/master/lib/config.js#L296

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue. For the sake of having all the discussion in one place, I'll restate what I mentioned in the Gitter channel earlier:

---

To clarify the original motivation for disallowing `..`: If I have a folder called `src/` and also a folder called `tests/`, and `tests/.eslintrc` contains an override for `"../src/**/*.js"`, then what should be the behavior of `eslint src/`? Either

* the override in `tests/.eslintrc` gets applied (which would require ESLint to somehow realize that `tests/.eslintrc` exists, despite the fact that it's not looking in tests), or
* the override in `tests/.eslintrc` does not get applied, in which case the config for `src/` would be different depending on whether `tests/` is linted at the same time.
Neither of these options seem good, so the solution was to disallow `..` in overrides.

The fact that overrides with `..` are also disallowed in configs when using the `--config` flag is a side-effect of the above dilemma and not a direct part of the design. That said, I'm not sure if there's a good way to improve it -- it seems like it could also be confusing for a directive to only be valid when the surrounding config file is used with `--config`, and not be valid otherwise. It also sees potentially confusing if `--config` changes how relative paths are resolved.

In other words, I don't think a solution for this would be as simple as allowing `..` and absolute paths in `overrides`, because that would introduce cases where ESLint behaves incorrectly/inconsistently when there's a config file in a sibling folder that has an override with `..`.
